Core/src/TrackFitting/GlobalChiSquareFitter.cpp

```diff
@@ -50,3 +50,97 @@ void Acts::Experimental::updateGx2fCovarianceParams(
 
   return;
 }
+
+void Acts::Experimental::addMeasurementToGx2fSumsBackend(
+    Gx2fSystem& extendedSystem,
+    const std::vector<BoundMatrix>& jacobianFromStart,
+    const Eigen::MatrixXd& covarianceMeasurement, const BoundVector& predicted,
+    const Eigen::VectorXd& measurement, const Eigen::MatrixXd& projector,
+    const Logger& logger) {
+  // First, w try to invert the covariance matrix. If the inversion fails, we
+  // can already abort.
+  const auto safeInvCovMeasurement = safeInverse(covarianceMeasurement);
+  if (!safeInvCovMeasurement) {
+    ACTS_WARNING("addMeasurementToGx2fSums: safeInvCovMeasurement failed.");
+    ACTS_VERBOSE("    covarianceMeasurement:\n" << covarianceMeasurement);
+    return;
+  }
+
+  // Create an extended Jacobian. This one contains only eBoundSize rows,
+  // because the rest is irrelevant. We fill it in the next steps.
+  // TODO make dimsExtendedParams template with unrolling
+  Eigen::MatrixXd extendedJacobian =
+      Eigen::MatrixXd::Zero(eBoundSize, extendedSystem.nDims());
+
+  // This part of the Jacobian comes from the material-less propagation
+  extendedJacobian.topLeftCorner<eBoundSize, eBoundSize>() =
+      jacobianFromStart[0];
+
+  // If we have material, loop here over all Jacobians. We add extra columns for
+  // their phi-theta projections. These parts account for the propagation of the
+  // scattering angles.
+  for (std::size_t matSurface = 1; matSurface < jacobianFromStart.size();
+       matSurface++) {
+    const BoundMatrix jac = jacobianFromStart[matSurface];
+
+    const ActsMatrix<eBoundSize, 2> jacPhiTheta =
+        jac * Gx2fConstants::phiThetaProjector;
+
+    // The position, where we need to insert the values in the extended Jacobian
+    const std::size_t deltaPosition = eBoundSize + 2 * (matSurface - 1);
+
+    extendedJacobian.block<eBoundSize, 2>(0, deltaPosition) = jacPhiTheta;
```

Comment:
Sonar is unhappy about this line. I think the message that this uses the comma operator is incorrect.
Maybe this would suppress this:

```suggestion
    extendedJacobian.template block<eBoundSize, 2>(0, deltaPosition) = jacPhiTheta;
```

Reply:
let's try. it compiles at least.